src/Components/src/Microsoft.AspNetCore.Components/Rendering/Renderer.cs

```diff
@@ -53,6 +53,13 @@ protected IComponent InstantiateComponent(Type componentType)
         protected int AssignRootComponentId(IComponent component)
             => AttachAndInitComponent(component, -1).ComponentId;
 
+        /// <summary>
+        /// Gets the current render tree for a given component.
+        /// </summary>
+        /// <param name="componentId">The id for the component.</param>
+        /// <returns>The <see cref="RenderTreeBuilder"/> representing the current render tree.</returns>
+        public RenderTreeBuilder GetCurrentRenderTree(int componentId) => GetRequiredComponentState(componentId).CurrrentRenderTree;
```

Comment:
Yeah, that sounds better, given the fact that the only thing we use are the frames.

Reply:
It it's reasonably possible, I think it would be good to avoid exposing `RenderTreeFrame` as public. This is something we've modified and tweaked about a billion times building components so far. I think it will cause us untoward pain if we make this a public contract.

Reply:
@rynowak The most likely way we'd avoid exposing `RenderTreeFrame` is by putting the `HtmlRenderer` into the `Microsoft.AspNetCore.Components` package directly (so this new method can be `internal`), which is something @javiercn was already suggesting for other reasons anyway.

My only concern about that is that client-side code doesn't need this functionality and we're sensitive about not bloating assemblies that are used on the client. But TBH I don't think it will be a problem because `HtmlRenderer`'s total added size is probably under 1KB, and in the long term we can look at enabling the linker for `Microsoft.AspNetCore.Components*` anyway.